src/utils/linked_list.rs

```diff
@@ -0,0 +1,383 @@
+    ///
+    /// let mut list = LinkedList::new();
+    /// let mut item = ListItem::new("Hello");
+    ///
+    /// let token = list.push(&mut item);
+    ///
+    /// {
+    ///   let item_option = list.pop();
+    ///   assert!(item_option.is_some());
+    ///   let item_ref = item_option.unwrap();
+    ///   assert_eq!(**item_ref, "Hello");
+    ///
+    ///   // The popped reference is dropped at the end of this block.
+    /// }
+    ///
+    /// unsafe {
+    ///   // Item was already popped from the list, and it's reference is dropped.
+    ///   // get_unchecked() can be safely used.
+    ///
+    ///   let item_ref = list.get_unchecked(token);
+    ///   assert_eq!(**item_ref, "Hello");
+    /// }
+    /// ```
+    pub unsafe fn get_unchecked(
+        &mut self,
+        token: ListItemToken<'list, T>,
+    ) -> &'list mut ListItem<'list, T> {
+        token.ptr.as_mut().unwrap()
+    }
+
+    /// Remove an element from this list
+    ///
+    /// Remove from the list an element represented by passed [token](ListItemToken).
+    ///
+    /// # Examples
+    ///
+    /// If the element is found in the list, exclusive borrow is returned to you.
+    ///
+    /// ```
+    /// use nrf_radio::utils::linked_list::{LinkedList, ListItem};
+    ///
+    /// let mut list = LinkedList::new();
+    /// let mut item = ListItem::new("Hello");
+    ///
+    /// let token = list.push(&mut item);
+    ///
+    /// let remove_result = list.remove(token);
+    /// assert!(remove_result.is_ok());
+    /// let item_ref = remove_result.unwrap();
+    /// assert_eq!(**item_ref, "Hello");
+    /// ```
+    ///
+    /// If the element is missing in the list, the token is returned to you.
+    ///
+    /// ```
+    /// use nrf_radio::utils::linked_list::{Error, LinkedList, ListItem};
+    ///
+    /// let mut list = LinkedList::new();
+    /// let mut item = ListItem::new("Hello");
+    ///
+    /// let token = list.push(&mut item);
+    ///
+    /// list.pop();
+    ///
+    /// let remove_result = list.remove(token);
+    /// let remove_error_option = remove_result.err();
+    /// assert!(remove_error_option.is_some());
+    /// let remove_error = remove_error_option.unwrap();
+    ///
+    /// match remove_error {
+    ///   Error::NotFound(token) => {
+    ///     // token can be stored for further usage
+    ///   }
+    ///   _ => panic!("Unexpected error"),
+    /// }
+    /// ```
+    pub fn remove(
+        &mut self,
+        token: ListItemToken<'list, T>,
+    ) -> Result<&'list mut ListItem<'list, T>, Error<'list, T>> {
+        enum PrevItem<'list, T> {
+            ListHead(*mut LinkedList<'list, T>),
+            ListItem(*mut ListItem<'list, T>),
+        }
+
+        // TODO: create an iterator instead?
+        let mut prev_item = PrevItem::ListHead(self);
+        let mut item_option: &mut Option<&'list mut ListItem<'list, T>> = &mut self.first;
+
+        // TODO: is this actually safe?
```

Comment:
You can use [miri](https://github.com/rust-lang/miri) to gain more confidence.

Reply:
I'll add miri to the checks. It's very interesting :)